app/jobs/europeana/feed_jobs/feed_job.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Europeana
+  module FeedJobs
+    class FeedJob < ActiveJob::Base
+      queue_as :cache
+
+      def perform(url, download_media = false)
+        @url = url
+        @download_media = download_media
+
+        @feed = ::Feedjira::Feed.fetch_and_parse(@url)
+
+        feed_cache_key = "feed/#{@url}"
+        cached_feed = Rails.cache.fetch(feed_cache_key)
+
+        if cached_feed.blank? || cached_feed.last_modified != @feed.last_modified
+          Rails.cache.write(feed_cache_key, @feed)
+          @updated = true
+        end
+      end
+
+      # Global nav uses some feeds, and is cached so needs to be expired when those
+      # feeds are updated.
+      # Cached pages need to be expired should they be using the updated feed.
+      after_perform do
```

Comment:
This does not belong here in the Engine.

I suggest:
1. Removing the `after_perform` block from here
2. Creating a new job class in europeana-portal-collections which sub-classes this one
3. Adding the `after_perform` block in to the app-specific job class